app/web/src/helpers/cadPackages/jsCad/jsCadController.ts

```diff
@@ -103,65 +135,31 @@ export const render: DefaultKernelExport['render'] = async ({
     const blob = new Blob([script], { type: 'text/javascript' })
     scriptWorker = new Worker(window.URL.createObjectURL(blob))
     let parameterDefinitions = []
-    scriptWorker.addEventListener('message', (e) => {
-      const data = e.data
+    scriptWorker.addEventListener('message', ({ data }) => {
       if (data.action == 'parameterDefinitions') {
         parameterDefinitions = data.data
       } else if (data.action == 'entities') {
         if (data.error) {
-          response = createUnhealthyResponse(new Date(), data.error)
+          workerHelper.resolver(createUnhealthyResponse(new Date(), data.error))
```

Comment:
Just passing the response into the `callResolve` equivalent so that the promise has the response inside.